src/installer/managed/Microsoft.NET.HostModel/MachO/MachObjectFile.cs

```diff
@@ -170,7 +170,7 @@ public static bool TryRemoveCodesign(MemoryMappedViewAccessor memoryMappedViewAc
     /// Removes the code signature load command and signature, and resizes the file if necessary.
     /// Returns true if the signature was removed, false otherwise.
     /// </summary>
-    public static bool TryRemoveCodesign(FileStream bundle)
+    public static void RemoveCodesign(FileStream bundle)
```

Comment:
nit: this can be:

```c#
    public static void RemoveCodesign(FileStream bundle)
    {
        // Windows doesn't allow a FileStream to be resized while the file is memory mapped, so we must dispose of the memory mapped file first.
        using MemoryMappedFile mmap = MemoryMappedFile.CreateFromFile(bundle, null, 0, MemoryMappedFileAccess.ReadWrite, HandleInheritability.None, true);
        using MemoryMappedViewAccessor accessor = mmap.CreateViewAccessor(0, 0, MemoryMappedFileAccess.ReadWrite);
        if (TryRemoveCodesign(accessor, out long? newLength))
        {
            bundle.SetLength(newLength.Value);
        }
    }
```

Reply:
Maybe in `else` case, we should throw `InvalidOperationException("Unable to remove Codesign from bundle.")`.

Reply:
I was getting failures on Windows when trying to resize the file length while the memory mapped file was still open, so the SetLength has to come after the `using` block.

The TryRemoveCodesign will only return false if the file isn't a Mach-O binary or there already isn't a signature present. We could throw if it's not a Mach-O, but I don't think it's a failure if the signature isn't present. Maybe we could rename both to `RemoveSignatureIfExists` instead?